app/Auth/Access/RegistrationService.php

```diff
@@ -72,6 +73,13 @@ public function findOrRegister(string $name, string $email, string $externalId):
             ];
 
             $user = $this->registerUser($userData, null, false);
+        } else if ($updateUser && ($user->name !== $name || $user->email !== $email)) {
+            $user->update([
+                'name'  => $name,
+                'email' => $email,
```

Comment:
The email should be user-unique so it might be wise to have a check before updating, but there becomes a question of what to do on error.

There's also a question of how this interacts with email confirmation and domain restriction. 
Don't want to complicate things thought so it might just be that those are left alone but we ensure there are warnings against the setting documentation about email confirmation/domain restriction.

Reply:
Is there really the need for email-verification in registrations from SSO-Services? In my oppinion this information must be verified by the identity-prodiver. I would suggest that in such cases a verified address is assumed.

Reply:
What do you think about removing email confirmation/domain restriction from SSO-Logins in general?

Reply:
> What do you think about removing email confirmation/domain restriction from SSO-Logins in general?

I am not willing to remove a feature that could be in use, just in the interest of making a PR easier to merge.
We can ignore these specific elements though and, as mentioned above, we'd just need to ensure these edge cases are in the documentation for the setting.

Reply:
This was not meant for making the PR easier to merge. I thought this would be a solution for preventing an inconsistent email-confirmation-feature in SAML-auth rather just "ignoring" it.

Now the question is how to resolve this, while maintaining consistency in function. How about a new config-parameter for email-domain-restrictions in SAML (and another parameter for every other auth-method)? So you can use domain-restriction on local-auth and disable it on saml-auth.

Reply:
As said above, I'd advise we ignore these specific elements in functionality and defer to documenting the implications instead, rather than increasing the surface area and scope of the PR.